### ISSUE TITLE
Dart/JS Sass compiler (to enable ARM64 support)

### ISSUE THREAD

**Issue by withinfocus** [statiqdev/Statiq.Framework#232]

**Comment by withinfocus** [comment]
I browsed around for a bit to see where this is headed and the SASS community is recommending https://github.com/Taritsyn/LibSassHost for .NET which has the earlier-referenced Dart version as well. While that's still marked as a preview version, it's getting more popular and is being used as a dependency in more and more projects. The current library is no longer being actively maintained, although maybe it would be reinvigorated at some point.

My question at this point is if you'd be willing to entertain a shift to the "SassHost" approach which will require some minor refactoring around the file import process and eventing. This seems more future-proof and could be done in phases without moving to Dart initially, perhaps most importantly because there are some breaking changes like the `OutputStyle` offerings there.

The catch here I suppose is that the Dart version -- as it's the supported one now -- is building with the desired ARM64 architecture, but you'd have to deal with the breaking changes.

**Comment by daveaglick** [comment]
I'd absolutely be open to dependency updates/changes - that seems inevitable as first Wyam and now Statiq comes up on 8 total years in service. Over such a long time span, and with such a large number of dependencies for various modules, it makes sense that there will be some drift and deprecation.

That said, my time is also very limited and I've been hesitant to work on parts that are currently functioning when there's still lots of greenfield work to be done. The Sass module works for many/most use cases today so taking the time to switch out it's compiler it's something I'm not likely to personally do for a while still.

TL;DR: I'd be thrilled to work with someone on a PR for this, but it'll still be a while before it rises high enough in the backlog that I'd work on it.

**Comment by withinfocus** [comment]
That was my way of saying "I may submit a PR" 😜 -- low priority for me too so I'll see what I can do in the near future.

**Comment by klasjersevi** [comment]
Any updates on where this is heading? Lib Sass is lacking more and more of the newer language features that Dart Sass provides, so It would be great to not have to maintain "legacy" Sass styling when using Statiq.

**Comment by daveaglick** [comment]
@klasjersevi It's partly dependent on finding a good library, and as far as I know there aren't any good alternatives right now. There's a wrapper around Dart Sass that I looked into, but because it relies on executing JavaScript, it's a little slow. And while we can actually run JavaScript code during generation (Statiq comes with a swappable JavaScript engine), it does limit the amount of integration that's available. In this case, we'd want the Sass engine to support a virtual file system since not all Statiq documents are available or expected from the file system.

Long story short: no additional work has been done on switching the Sass engine (by me at least), and until I find a more significant chunk of time, it's not critical on my own ranking of backlog issues at the moment.

**Comment by withinfocus** [comment]
Per the https://github.com/xoofx/SharpScss/issues/19 closure if the package is upgraded here we will get arm64 support!
